docs/service-bundle.md

```diff
@@ -0,0 +1,230 @@
+
+```
+apiVersion: v1
+data:
+  fields: eyJEQl9OQU1FIjogImZvbyIsICJEQl9QQVNTV09SRCI6ICJsZXRtZWluIiwgIkRCX1RZUEUiOiAicG9zdGdyZXMiLCAiREJfUE9SVCI6ICI1NDMyIiwgIkRCX1VTRVIiOiAiYmFyIiwgIkRCX0hPU1QiOiAicG9zdGdyZXNxbCJ9
+kind: Secret
+metadata:
+  creationTimestamp: 2018-02-26T18:34:50Z
+  name: apb-874ae78d-acfe-4181-b064-124d6e13c592
+  namespace: dh-postgresql-apb-prov-md5xt
+  resourceVersion: "11846"
+  selfLink: /api/v1/namespaces/dh-postgresql-apb-prov-md5xt/secrets/apb-874ae78d-acfe-4181-b064-124d6e13c592
+  uid: ba79e27a-1b23-11e8-aebf-94b96c9ea8ca
+type: Opaque
+```
+
+The value for “fields” is a base64-encoded JSON document whose keys and values
+represent any data the APB wants to return to the broker and ultimately to a
+client that retrieves a binding.
+
+__Provision Credentials__ - If a service bundle saves such a secret during the
+provision operation, the broker saves the passed data as “provision
+credentials”. That data gets passed into all further actions as part of the
+JSON document under the key ``_apb_provision_creds``, as a sub-document. As such,
+it can be useful for making root-level credentials created during provision
+available to other operations that need to use them. If a request for non-async
+binding is made and the bundle does not run, any “provision credentials” that
+were saved will be returned to the client as binding credentials.
+
+__Binding Credentials__ - If a service bundle saves such a secret during the bind
+operation, the broker saves the passed data as “binding credentials”. These
+credentials get returned to the client that initiated that specific binding
+operation.
+
+## Runtime
+
+The container will be run in a pod by itself with ``restartPolicy: Never``. The
+pod will run in a sandbox namespace created by the broker for that specific
+operation only. The namespace is not intended to persist, and in normal
+operation it will be deleted as soon as the pod completes. The broker can
+optionally be configured to retain the namespace for debugging purposes.
+
+The pod is run with a service account granted access to the ``POD_NAMESPACE`` and
+to the “namespace” that the pod is to deploy to. Networks between the two
+namespaces are tied together so that you can reliably find your services in the
+target namespace. That only works for
+[NetworkPolicy](https://kubernetes.io/docs/concepts/services-networking/network-policies/)
+or for multi tenant openshift-sdn.
+
+### Proxy
+
+If there are proxy settings for the broker, they will be forwarded to the pod
+by setting environment variables for ``HTTP_PROXY``, ``HTTPS_PROXY``,
+``NO_PROXY``, ``http_proxy``, ``https_proxy``, and ``no_proxy``.
+
+No action is taken to apply proxy settings to pods that a service bundle
+creates. Each service bundle is repsonsible for configuring its provisioned
+resources to use a proxy as appropriate.
+
+### Service Account
+
+Details of a kubernetes service account are mounted at
+``/var/run/secrets/kubernetes.io/serviceaccount``
```

Comment:
https://kubernetes.io/docs/admin/service-accounts-admin/ I think we may want to avoid getting into the specifics here.

Reply:
The `apb-base` has this generic [kube-config](https://github.com/ansibleplaybookbundle/apb-base/blob/master/files/opt/apb/.kube/config) that it allows it to simply use the service account that was given.